xds/src/main/java/com/linecorp/centraldogma/xds/group/v1/XdsGroupService.java

```diff
@@ -0,0 +1,119 @@
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.centraldogma.xds.group.v1;
+
+import static com.linecorp.centraldogma.server.internal.admin.auth.AuthUtil.currentAuthor;
+import static com.linecorp.centraldogma.server.internal.api.RepositoryServiceUtil.createRepository;
+import static com.linecorp.centraldogma.server.internal.api.RepositoryServiceUtil.removeRepository;
+import static com.linecorp.centraldogma.xds.internal.ControlPlanePlugin.XDS_CENTRAL_DOGMA_PROJECT;
+
+import com.google.protobuf.Empty;
+
+import com.linecorp.centraldogma.common.RepositoryExistsException;
+import com.linecorp.centraldogma.server.command.CommandExecutor;
+import com.linecorp.centraldogma.server.metadata.MetadataService;
+import com.linecorp.centraldogma.server.storage.project.Project;
+import com.linecorp.centraldogma.server.storage.project.ProjectManager;
+import com.linecorp.centraldogma.xds.group.v1.XdsGroupServiceGrpc.XdsGroupServiceImplBase;
+
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.StreamObserver;
+
+/**
+ * An {@link XdsGroupServiceImplBase} implementation that provides methods to manage XDS groups.
+ */
+public final class XdsGroupService extends XdsGroupServiceImplBase {
+
+    private final ProjectManager projectManager;
+    private final CommandExecutor commandExecutor;
+    private final MetadataService mds;
+
+    /**
+     * Creates a new instance.
+     */
+    public XdsGroupService(ProjectManager projectManager, CommandExecutor commandExecutor) {
+        this.projectManager = projectManager;
+        this.commandExecutor = commandExecutor;
+        mds = new MetadataService(projectManager, commandExecutor);
+    }
+
+    @Override
+    public void createGroup(CreateGroupRequest request,
+                            StreamObserver<Group> responseObserver) {
+        final String groupName = request.getGroup().getName();
+        final String name = removePrefix("groups/", groupName);
+        if (projectManager.get(XDS_CENTRAL_DOGMA_PROJECT).repos().exists(name)) {
+            throw alreadyExistsException(name);
+        }
+        createRepository(commandExecutor, mds, currentAuthor(), XDS_CENTRAL_DOGMA_PROJECT, name)
+                .handle((unused, cause) -> {
+                    if (cause != null) {
+                        if (cause instanceof RepositoryExistsException) {
+                            responseObserver.onError(alreadyExistsException(name));
+                        } else {
+                            responseObserver.onError(
+                                    Status.INTERNAL.withCause(cause).asRuntimeException());
```

Comment:
Note: I'm curious on how we'll deal with handling exceptions later on as sometimes we perform special operations like in `ReplicationLagTolerantCentralDogma` for `RevisionNotFoundException`.

For now, I assume creating a repository and getting the repository immediately won't necessary succeed if the get is from a different replica.

Anyways, I guess the current approach is fine for now

Reply:
That's a good question. I will add the gRPC exception handler for the xDS services after I all set up the services.